internal/application/manager.go

```diff
@@ -171,9 +171,22 @@ func (m *dataManager) CancelRecording() error {
 }
 
 // RecordingStatus returns the status of the current recording session
-func (m *dataManager) RecordingStatus() (*dtos.RecordStatus, error) {
-	//TODO implement me using TDD
-	return nil, errors.New("not implemented")
+func (m *dataManager) RecordingStatus() *dtos.RecordStatus {
+	m.recordingMutex.Lock()
+	defer m.recordingMutex.Unlock()
+
+	status := &dtos.RecordStatus{}
+
+	if m.recordingStartedAt != nil {
+		status.InProgress = true
+		status.Duration = time.Since(*m.recordingStartedAt)
+		status.EventCount = m.eventCount
+	} else if m.recordedData != nil {
+		status.Duration = m.recordedData.Duration
+		status.EventCount = len(m.recordedData.Events)
+	}
```

Comment:
if both `m.recordingStartedAt and *m.recordedData are nil, what is the RecordStatus?  maybe return nil in this case?

Reply:
Status is no recorded data which is represented by the empty RecordStatus of
```
{
    "inProgress": false,
    "eventCount": 0,
    "duration": 0
}
```

Reply:
ok, that makes sense then